src/manageiq_client/api.py

```diff
@@ -616,11 +616,33 @@ def __call__(self, *args, **kwargs):
             self.api.response = action_response
         return outcome
 
+    def _get_entity_from_href(self, result):
+        href_result = result['href']
+        href_match = None
+
+        # If the "href" value in result doesn't match the current collection,
+        # try to find the collection that the "href" refers to.
+        if self.collection._href not in href_result:
```

Comment:
@mkoura do you think this can be a little more strict? e.g. something like `not self.collection._href.startswith(href_result)`

Reply:
Agree, this is better.